tools/deploy_firmware.py

```diff
@@ -0,0 +1,288 @@
+
+    baudrate = 115200
+    """The baudrate of the serial connection to the device."""
+
+    def __init__(self, device: str, name: str, image: str, username: str = "root"):
+        """
+
+        Parameters
+        -----------
+        device: str
+            The name of the platform (example: netgear-rax40).
+        name: str
+            The name of the device (it should ne reachable through ssh without a password).
+        image: str
+            The name of the image that can be used to upgrade the device.
+        username: str, optional
+            The username to use when connecting to the device over SSH.
+        """
+        self.device = device
+        self.name = name
+        self.image = image
+        self.username = username
+
+        self.scriptdir = os.path.join(os.path.dirname(os.path.realpath(__file__)), "..")
+        self.rootdir = os.path.realpath(os.path.join(self.scriptdir))
+        self.artifacts_dir = os.path.join(self.rootdir, "build/{}".format(self.device))
+        """The directory where artifacts are stored. It's expected to contain
+        prplwrt-version, and the image file."""
+
+    def set_prompt(self, pexp: pexpect):
+        """Set the prompt to `shell_prompt` on a pexpect object."""
+        pexp.sendline("export PS1='{}'".format(self.shell_prompt))
+        pexp.sendline("")
+
+    def read_remote_version(self) -> List[str]:
+        """Read prplwrt-version on a remote device
+
+        Returns
+        -------
+        List[str]
+            The content of the prplwrt-version file on the device as a list of strings.
+        """
+        with pexpect.spawn("ssh {}@{}".format(self.username, self.name)) as shell:
+            shell.expect("{}@".format(self.username))
+            shell.sendline("cat /etc/prplwrt-version")
+            shell.expect("cat /etc/prplwrt-version")
+            shell.expect("{}@".format(self.username))
+            return shell.before.decode().split()
+
+    def read_local_version(self) -> List[str]:
+        """Reads the local prplwrt-version.
+
+        The version file is read from the artifacts (see `artifacts_dir`).
+
+        Returns
+        -------
+        List[str]
+            The content of the local prplwrt-version as a list of strings.
+        """
+        with open(os.path.join(self.artifacts_dir, "prplwrt-version")) as version_file:
+            return version_file.read().splitlines()
+
+    def needs_upgrade(self):
+        """Check if a device needs to be updated
+
+        The check is done by comparing prplwrt-version on the target
+        with a local one (see `read_local_version).
+        """
+        local_version = self.read_local_version()
+        remote_version = self.read_remote_version()
+        diff = difflib.unified_diff(local_version, remote_version,
+                                    fromfile='file1', tofile='file2', lineterm='')
+        diff_str = '\n'.join(diff)
+        print(diff_str)
```

Comment:
```suggestion
        if diff_str:
            print(diff_str)
```
Otherwise you get an empty line.